rotkehlchen/chain/optimism/modules/velodrome/velodrome_cache.py

```diff
@@ -0,0 +1,217 @@
+        existing_pools: list[ChecksumEvmAddress],
+) -> Optional[list[VelodromePoolData]]:
+    """
+    Queries velodrome data from chain from the Velodrome Finance LP Sugar v2 contract.
+    If new pools are found their tokens are created and the pools are returned.
+    (Find more details for the contract here: https://github.com/velodrome-finance/sugar).
+
+    May raise:
+    - RemoteError if there is an error connecting with the remote source of data
+    """
+    data_contract = inquirer.contracts.contract(VELODROME_SUGAR_V2_CONTRACT)
+    pool_data = []
+    offset = 0
+    limit = 200
+    pool_data_chunk: list[dict] = []
+    while len(pool_data_chunk) == limit or (pool_data_chunk == [] and offset == 0):
+        pool_data_chunk = data_contract.call(
+            node_inquirer=inquirer,
+            method_name='all',
+            arguments=[limit, offset, ZERO_ADDRESS],  # the address argument here is needed by the design of this contract interface to cover other user cases its developers had in mind, but it isn't needed for this particular call in which we only want to get the pools. So we can just use any address.  # noqa: E501
+        )
+        pool_data.extend(pool_data_chunk)
+        offset += limit
+
+    pools = []
+    for pool in pool_data:
+        try:
+            pool_address = deserialize_evm_address(pool[0])
+        except DeserializationError as e:
+            log.error(
+                f'Skipping velodrome pool {pool[0]}. Could not deserialize it while '
+                f'decoding it. Additional information: {e}',
+            )
+            continue
+
+        if pool_address in existing_pools:
+            continue
+
+        try:
+            token0, token1 = deserialize_evm_address(pool[5]), deserialize_evm_address(pool[8])
+            gauge_address = deserialize_evm_address(pool[11])
+        except DeserializationError as e:
+            log.error(
+                f'Skipping velodrome pool {pool[0]}. Could not deserialize an evm address while '
+                f'decoding it. Additional information: {e}',
+            )
+            continue
+
+        skip_pool = False
+        for token in (token0, token1):  # create the tokens for the new pools
+            try:
+                get_or_create_evm_token(
+                    userdb=inquirer.database,
+                    evm_address=token,
+                    chain_id=ChainID.OPTIMISM,
+                    evm_inquirer=inquirer,
+                    seen=TokenSeenAt(description='Querying velodrome pools'),
+                )
+            except NotERC20Conformant as e:
+                skip_pool = True
+                log.error(
+                    f'Skipping token {token} because it is not a valid ERC20 token. {e}',
+                )
+
+        if skip_pool is True:
+            log.error(f'Skipping velodrome pool {pool_address} because one of its tokens is not a valid ERC20')  # noqa: E501
+            continue
```

Comment:
why this and not just add the continue in the except part?

Reply:
It will continue the token for loop, not the pool loop.